baseline-network-linux-hld.md

```diff
@@ -0,0 +1,162 @@
+
+- recipes for the basic kernel mentioned above and for platform support
+- class and image recipe for an ONIE installable image
+- machine definitions as generic OS-based
+
+### Kernel recipe and configuration handling
+
+The kernel configuration will be done via
+https://docs.yoctoproject.org/kernel-dev/advanced.html to allow large
+flexibility in how other projects using the layer can customize the kernel.
+
+TBD: should we have patches or branches with backport commits? Maybe both?
+
+- patches make required changes due to upstream changes more obvious (provide a
+“conflict diff”) and easier to review
+- branches are closer to the upstream approach, but may be harder to track
+  changes needed due to upstream changes
+
+### Platform support recipes
+
+To allow easier extension by vendors or other users, the platform support will
+be split into multiple subpackages:
+
+- a base package for the userspace utilities and API libraries
+- one or multiple support packages for the platforms
+
+### ASIC/dataplane support
+
+ASIC/dataplane support is out of scope for now, but may be provided by a future
+project or extension.
+
+The only exception are switchdev-enabled platforms like Mellanox or Marvell
+Prestera, where the ASIC support is already provided by the upstream Linux kernel.
+
+### Targeted Kernel and Yocto versions
+
+Target Yocto version is scarthgarp, which is the next LTS release, planned for
+April 2024.
+
+The initial Kernel version target could be 5.15 (minimum kernel version of Yocto)
+6.6 (the latest LTS version of the Linux kernel). The kernel patch version will be a moving target, and regularly
+updated (ideally weekly).
+
+Since there will be many out-of-tree drivers for the platform support, required
+API updates should be done via [coccinelle semantic patches](https://coccinelle.gitlabpages.inria.fr/website/). This should ensure that the
+amount of work needed for keeping drivers building and working with newer
+kernels stays manageable.
+
+These will also help vendors quickly updating their drivers when submitting new
+platforms based on older releases.
+
+### DentOS as a layer extending the base platform support layer
+
+The base platform support will be already provided by the base platform layer
+(i.e. custom hardware drivers, platform initialization, etc).
+
+Since DentOS uses a customized version of the prestera driver that contains
+several features not included in the upstream driver, the custom prestera
+driver will be provided as an external kernel module package, while the
+in-kernel driver will be disabled.
+
+## Milestones
+
+- M1: initial layer with kernel and hand-selected/-crafted platform support,
+  booting a ramdisk kernel
+- M2: refined kernel configuration via kernel-meta, import of various platforms
+  via scripts from SONiC, initial installer
+- M3: import missing platforms from ONL and DentOS, ONLP adapter(?), extensible
+  installer
+- M4: get accepted as official layer
+
+## Future Extensions
```

Comment:
Even with the "future extensions", proposal does not sound like there is a roadmap to provide a full network switch support including data and control planes for an OS.